astropy/table/_column_mixins.pyx

```diff
@@ -56,10 +57,14 @@ cdef inline object base_getitem(object self, object item, item_getter getitem):
     if (<ndarray>self).ndim > 1 and isinstance(item, INTEGER_TYPES):
         return self.data[item]
 
+    dtype = (<ndarray>self).dtype
+    if dtype.kind == 'V' and isinstance(item, STRING_TYPES):
+        return self.data[item]
+
     value = getitem(self, item)
 
     try:
-        if value.dtype.char == 'S' and not value.shape:
+        if not value.shape and dtype.char == 'S':
```

Comment:
@mhvk - excellent suggestions. With those in the latest version the benchmark above is ~125 ns in this PR vs 142 ns in main. So it is actually faster now. I expect that if we did the same tweaks in main it would jump ahead to be faster still, but never mind that.

Reply:
Very nice to hear! I think this is all good then but it still needs a changelog entry...